public/app/api/README.md

```diff
@@ -0,0 +1,160 @@
+| filterEndpoints | The `operationId` of the particular route you want to work with. You can check the available operationIds in the specific group's spec file. As seen in the `migrate-to-cloud` one, it is an array                                                                                        |
+|  tag            | Must be set to `true`, to automatically attach tags to endpoints. This is needed for proper cache invalidation. See more info in the [official documentation](https://redux-toolkit.js.org/rtk-query/usage/automated-refetching#:~:text=RTK%20Query%20uses,an%20active%20subscription.).  |
+|  hooks          | Must be set to `false` since we only want to export the hooks from the index file, not the generated one.                                                                                                                                                                                 |
+
+<br/>
+> More info in [Redux Toolkit](https://redux-toolkit.js.org/rtk-query/usage/code-generation#simple-usage)
+
+In our example, the information added will be:
+
+```jsx
+'../public/app/features/dashboard/api/endpoints.gen.ts': {
+    apiFile: '../public/app/features/dashboard/api/baseAPI.ts',
+    schemaFile: '../data/openapi/dashboard.grafana.app-v0alpha1.json',
+    apiImport: 'baseAPI',
+    filterEndpoints: ['createDashboard'],
+    tag: true,
+    hooks: false,
+},
+```
+
+### 5. Run the API Client script
+
+Then, we are ready to run the script to create the API client:
+
+```jsx
+yarn generate-apis
+```
+
+This will create an `endpoints.gen.ts` file in the path specified in the previous step.
+
+### 6. Create the index file for your hooks
+
+In the same `api` folder where the `endpoints.gen.ts` file has been saved, you have to create an index file from which you can import the types and hooks needed. By doing this, we selectively export hooks/types from `endpoints.gen.ts`.
+
+In our case, the dashboard index will be like:
+
+```jsx
+import { generatedAPI } from './endpoints.gen';
+
+export const dashboardAPI = generatedAPI;
+export const { useCreateDashboardMutation} = dashboardAPI;
+// eslint-disable-next-line no-barrel-files/no-barrel-files
+export { type Dashboard } from './endpoints.gen';
+
+```
+
+There are some use cases where the hook will not work, and that is a clue to see if it needs to be modified. The hooks can be tweaked by using `enhanceEndpoints`.
+
+```jsx
+export const dashboardsAPI = generatedApi.enhanceEndpoints({
+  endpoints: {
+    // Need to mutate the generated query to set the Content-Type header correctly
+    createDashboard: (endpointDefinition) => {
+      const originalQuery = endpointDefinition.query;
+      if (originalQuery) {
+        endpointDefinition.query = (requestOptions) => ({
+          ...originalQuery(requestOptions),
+          headers: {
+            'Content-Type': 'application/merge-patch+json',
+          },
+        });
+      }
+    },
```

Comment:
This example is a bit weird, why do we use patch content type for a POST request? Did you maybe mean `updateDashboard`?